From 1b2dcb31eefbcec2825006bf5a16e11492d9e61e Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Fri, 18 Jan 2019 20:46:07 -0800
Subject: [PATCH] Update the muttrc man page with added commands.

Add the "color compose", index-format-hook, and echo commands.

Add the HMS relative date units.
---
 doc/muttrc.man.head | 27 +++++++++++++++++++++++++--
 1 file changed, 25 insertions(+), 2 deletions(-)

diff --git a/doc/muttrc.man.head b/doc/muttrc.man.head
index 92b40935..e1dc2b10 100644
--- a/doc/muttrc.man.head
+++ b/doc/muttrc.man.head
@@ -223,6 +223,7 @@ specify multiple maps, put only a comma between the maps.
 .nf
 \fBcolor\fP \fIobject\fP \fIforeground\fP \fIbackground\fP [ \fIregexp\fP ]
 \fBcolor\fP index \fIforeground\fP \fIbackground\fP [ \fIpattern\fP ]
+\fBcolor\fP compose \fIcomposeobject\fP \fIforeground\fP \fIbackground\fP
 \fBuncolor\fP index \fIpattern\fP [ \fIpattern\fP ... ]
 .fi
 .IP
@@ -243,6 +244,10 @@ objects allow you to restrict the colorization to a regular
 expression.  The \fBindex\fP object permits you to select colored
 messages by pattern.
 .IP
+Valid composeobjects include
+.BR header ", " security_encrypt ", " security_sign ", "
+.BR security_both ", " security_none .
+.IP
 Valid colors include:
 .BR white ", " black ", " green ", " magenta ", " blue ", "
 .BR cyan ", " yellow ", " red ", " default ", " color\fIN\fP .
@@ -361,6 +366,16 @@ You may use multiple
 \fBcrypt-hook\fPs with the same \fIregexp\fP; multiple matching
 \fBcrypt-hook\fPs result in the use of multiple \fIkey-id\fPs for
 a recipient.
+.TP
+\fBindex-format-hook\fP \fIname\fP [\fB!\fP]\fIpattern\fP \fIformat-string\fP
+This command is used to inject format strings dynamically into
+$index_format based on pattern matching against the current message.
+.IP
+The $index_format expando \fI%@name@\fP specifies a placeholder for
+the injection. Index-format-hooks with the same \fIname\fP are matched
+using \fIpattern\fP against the current message. Matching is done in
+the order specified in the .muttrc, with the first match being
+used. The hook's \fIformat-string\fP is then substituted and evaluated.
 .PP
 .nf
 \fBopen-hook\fP \fIregexp\fP "\fIcommand\fP"
@@ -464,6 +479,11 @@ These commands allow the user to modify the list of allowed header
 fields in a \fImailto:\fP URL that Mutt will include in the
 the generated message.  By default the list contains only
 \fBsubject\fP and \fBbody\fP, as specified by RFC2368.
+.TP
+\fBecho\fP \fImessage\fP
+Prints \fImessage\fP to the message window. After printing the
+message, echo will pause for the number of seconds specified by
+$sleep_time.
 .SH PATTERNS
 .PP
 In various places with mutt, including some of the above mentioned
@@ -734,8 +754,11 @@ followed by a positive number, followed by one of the unit
 characters
 .BR y ,
 .BR m ,
-.BR w ", or"
-.BR d .
+.BR w ,
+.BR d ,
+.BR H ,
+.BR M ", or"
+.BR S .
 .B >
 matches dates which are older than the specified amount of time, an
 offset which begins with the character
-- 
2.20.1

